docs: v1.1.2 (#5850)
langflow-ai/langflow: docs/docs/Tutorials/tutorials-memory-chatbot.md

![](/img/starter-flow-memory-chatbot.png)

> This flow adds a **Chat Memory** component to the Basic Prompting flow. This component retrieves previous messages and sends them to the **Prompt** component to fill a part of the **Template** with context.

Reviewer: ```suggestion
This flow adds a **Chat Memory** component to the Basic Prompting flow.
This component retrieves previous messages and sends them to the **Prompt** component to fill a part of the **Template** with context.
```

To examine the template, clicking the **Template** field in the **Prompt** component.
The **Prompt** tells the **OpenAI model** component how to respond to input.

```plain
You are a helpful assistant that answers questions.

Use markdown to format your answer, properly embedding images and urls.

History:

{memory}
```

The `{memory}` code in the prompt creates a new input port in the component called **memory**.
The **Chat Memory** component is connected to this port to store chat messages from the **Playground**.

This gives the **OpenAI** component a memory of previous chat messages.

## Run the memory chatbot flow

1. Open the **Playground**.
2. Type multiple questions. For example, try entering this conversation:

```plain
Hi, my name is Luca.
Please tell me about PostgreSQL.
What is my name?
What is the second subject I asked you about?
```

> The chatbot should remember your name and previous questions.

Reviewer: ```suggestion
The chatbot remembers your name and previous questions.
```

> 3. To view the **Message Logs** pane, click <Icon name="Ellipsis" aria-label="Horizontal ellipsis" /> and then click **Message Logs**.

Reviewer: ```suggestion
3. To view the **Message Logs** pane, click <Icon name="Ellipsis" aria-label="Horizontal ellipsis" />, and then click **Message Logs**.
```
